ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -289,13 +303,16 @@ newWallet ::
     -> WalletConfig
     -> WalletFace
     -> IO PassPhrase
+    -> (ConfirmationType -> IO Bool)
     -> Maybe WalletName
     -> Maybe Byte
     -> IO [Text]
-newWallet pwl walletConfig face getPassTemp mbWalletName mbEntropySize = do
+newWallet pwl walletConfig face getPassTemp waitUiConfirm mbWalletName mbEntropySize = do
   pp <- getPassTemp
   let entropySize = fromMaybe (wcEntropySize walletConfig) mbEntropySize
   mnemonic <- generateMnemonic entropySize
+  confirmed <- waitUiConfirm $ ConfirmMnemonic mnemonic
+  when (not confirmed) $ throwM WGFailedUnconfirmedMnemonic
```

Comment:
`hlint` suggests `unless confirmed $` here. Actually you can use `unlessM (walletUiConfirm ...) $ throwM`  and avoid extra variable completely if you prefer so.